### PR TITLE
Update scheduledQueryRule_API.json To use the right error contract

### PR DESCRIPTION
**Pull request by adklinge**
Fixed Error contract to fit the real structure

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [x] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://msazure.visualstudio.com/One/_workitems/create/Scenario?templateId=d1c215ec-5d6a-4cb3-a193-95ddd0a1ba96&ownerId=85bb92fb-21fe-41ca-9309-47f2c5f57515). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by adklinge** [comment]
@kairu-ms - The breaking changes errors are all due to the change in the schema of the error message.
This change (derived from s360 violations) is done only in the swagger, to align it with the service's actual error response schema. Therefor, there is no real breaking change here. 
Pease let me know how can we proceed from here.
Thank you! :)

**Comment by adklinge** [comment]
@kairu-ms - as for the failures in Azure Sdk for .net, I am not sure how my PR changes effected the changes in :
sdk/monitor/Microsoft.Azure.Management.Monitor/src/Generated/IMonitorManagementClient.cs

It does not seem related. Could you please take a look and advice how to resolve it ?
Thank you

**Comment by kairu-ms** [comment]
> @kairu-ms Kai Ru FTE - The breaking changes errors are all due to the change in the schema of the error message.
> This change (derived from s360 violations) is done only in the swagger, to align it with the service's actual error response schema. Therefor, there is no real breaking change here.
> Pease let me know how can we proceed from here.
> Thank you! :)

@adklinge It does have a breaking change. Because your error response body has been changed from
```json
{"code": "errorCode", "message": "errorMessage"}
```
to
```json
{"error": {"code": "errorCode", "message": "errorMessage"}}
```

**Comment by adklinge** [comment]
> > @kairu-ms Kai Ru FTE - The breaking changes errors are all due to the change in the schema of the error message.
> > This change (derived from s360 violations) is done only in the swagger, to align it with the service's actual error response schema. Therefor, there is no real breaking change here.
> > Pease let me know how can we proceed from here.
> > Thank you! :)
> 
> @adklinge It does have a breaking change. Because your error response body has been changed from
> 
> ```json
> {"code": "errorCode", "message": "errorMessage"}
> ```
> 
> to
> 
> ```json
> {"error": {"code": "errorCode", "message": "errorMessage"}}
> ```

Hi @kairu-ms, the server has always returned :
 {"error": {"code": "errorCode", "message": "errorMessage"}}

It is just the swagger document that was not aligned and showed : 
{"code": "errorCode", "message": "errorMessage"}

We are not changing anything in the service response, therefor there will be no customer breaking change.

**Comment by kairu-ms** [comment]
> Hi @kairu-ms Kai Ru FTE, the server has always returned :
> {"error": {"code": "errorCode", "message": "errorMessage"}}
> 
> It is just the swagger document that was not aligned and showed :
> {"code": "errorCode", "message": "errorMessage"}
> 
> We are not changing anything in the service response, therefor there will be no customer breaking change.

Hi @adklinge. It is a breaking change. You should follow [Breaking Change Policy](http://aka.ms/bcforapi).  
@akning-ms should I approve this?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2018-04-16/scheduledQueryRule_API.json
Comment (adklinge) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2018-04-16/scheduledQueryRule_API.json:539]: This is a fix to s360 violation : 
https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=D6E5FC34-0663-472E-952E-8B9F5C0C8D17
When ConditionalOperator is not specified by PUT,  the service populates it with the default value of 'GreaterThanOrEqual'. 
Therefor, this is not a breaking (we do not change the service current behavior) change but an alignment of the swagger.

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2018-04-16/scheduledQueryRule_API.json
Comment (adklinge) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2018-04-16/scheduledQueryRule_API.json:552]: This is a fix to s360 violation :
https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=D6E5FC34-0663-472E-952E-8B9F5C0C8D17
When "MetricTriggerType" is not specified by PUT, the service populates it with the default value of 'Consecutive'.
Therefor, this is not a breaking (we do not change the service current behavior) change but an alignment of the swagger.

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2020-05-01-preview/scheduledQueryRule_API.json
Comment (adklinge) [inline specification/monitor/resource-manager/Microsoft.Insights/preview/2020-05-01-preview/scheduledQueryRule_API.json:376]: This is a fix to s360 violation :
https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=2EAB7F16-67CA-44F1-8877-39B8404F881F

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2020-05-01-preview/scheduledQueryRule_API.json
Comment (adklinge) [inline specification/monitor/resource-manager/Microsoft.Insights/preview/2020-05-01-preview/scheduledQueryRule_API.json:336]: This is a fix to s360 violation :
https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=2EAB7F16-67CA-44F1-8877-39B8404F881F

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2020-05-01-preview/scheduledQueryRule_API.json
Comment (adklinge) [inline specification/monitor/resource-manager/Microsoft.Insights/preview/2020-05-01-preview/scheduledQueryRule_API.json:417]: This is a fix to s360 violation :
https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=C09C7410-CA54-4450-96BF-BF46F64F2190

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2018-04-16/scheduledQueryRule_API.json
Comment (adklinge) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2018-04-16/scheduledQueryRule_API.json:123]: This is a fix to s360 violation :
https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=60BD29A9-717D-4319-A4F4-6DD8D39EE288

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2020-05-01-preview/scheduledQueryRule_API.json
Comment (adklinge) [inline specification/monitor/resource-manager/Microsoft.Insights/preview/2020-05-01-preview/scheduledQueryRule_API.json:338]: In cases where the rule was migrated from the Legacy Log Analytic API

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2018-04-16/scheduledQueryRule_API.json
Comment (akning-ms) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2018-04-16/scheduledQueryRule_API.json:37]: If this is for fix Linter S360 issue. as it cause breaking change. you can send a exception request to suppress it.  
Pls revert this change

**Comment by akning-ms** [comment]
> > Hi @kairu-ms Kai Ru FTE, the server has always returned :
> > {"error": {"code": "errorCode", "message": "errorMessage"}}
> > It is just the swagger document that was not aligned and showed :
> > {"code": "errorCode", "message": "errorMessage"}
> > We are not changing anything in the service response, therefor there will be no customer breaking change.
> 
> Hi @adklinge. It is a breaking change. You should follow [Breaking Change Policy](http://aka.ms/bcforapi).
> @akning-ms should I approve this?

If these changes is align to service implementation to fix S360 correctness issue, you can follow below step to get approval from BreakingChange review board. it very likely get approval.  but for Linter S360 issue. if it cause Breaking change. you don't have to fix it. pls send a exception request to suppress it via https://aka.ms/swaggerkpiex. @adklinge 
https://github.com/Azure/azure-rest-api-specs/pull/11898#issuecomment-737127888

**Review by kairu-ms — APPROVED** [review]

**Comment by adklinge** [comment]
Hi @kairu-ms , thank you for approving.
I can not merge the PR as it says :
image


Please help. 
Thank you

**Comment by kairu-ms** [comment]
Hi @adklinge, please run prettify for you files.

**Comment by kairu-ms** [comment]
I found the same change in PR https://github.com/Azure/azure-rest-api-specs/pull/12033. Which one would your prefer to merge?

**Comment by kairu-ms** [comment]
@jianyexi Please help with linter issues. Thanks.

**Comment by jianyexi** [comment]
> @jianyexi Please help with linter issues. Thanks.
The 'SubscriptionIdParameter' in 
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/monitor/resource-manager/Microsoft.Insights/preview/2017-05-01-preview/subscriptionDiagnosticsSettings_API.json#L323 has conflict with the https://github.com/Azure/azure-rest-api-specs/blob/master/specification/monitor/resource-manager/Microsoft.Insights/stable/2015-04-01/autoscale_API.json#L853 ， please unify its definition  . Since the conflict already existed in the master branch ,  please ask @akning-ms to  force merge  after fixing the conflict

**Comment by adklinge** [comment]
> > @jianyexi Please help with linter issues. Thanks.
> > The [SubscriptionIdParameter](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/monitor/resource-manager/Microsoft.Insights/preview/2017-05-01-preview/subscriptionDiagnosticsSettings_API.json#L323) has conflict with the [SubscriptionIdParameter](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/monitor/resource-manager/Microsoft.Insights/stable/2015-04-01/autoscale_API.json#L853) ， please unify its definition (use the second one) . Since the conflict already existed in the master branch ,  please ask @akning-ms to  force merge  after fixing the conflict

@jianyexi, @kairu-ms - the file you ask me to modify is owned by another resource provider (diagnosticSettings), it does not make any sense to force us to fix other RPs in order to be able to maintain our own. Can we skip this somehow ?

**Comment by adklinge** [comment]
> I found the same change in PR #12033. Which one would your prefer to merge?

I would prefer merging this one

**Comment by adklinge** [comment]
> Hi @adklinge, please run prettify for you files.

@kairu-ms could you please specify what is the command I should run exactly ?

**Comment by kairu-ms** [comment]
> > Hi @adklinge, please run prettify for you files.
> 
> @kairu-ms could you please specify what is the command I should run exactly ?

You should prettify the json files in your IDE.

**Comment by jianyexi** [comment]
> > > @jianyexi Please help with linter issues. Thanks.
> > > The [SubscriptionIdParameter](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/monitor/resource-manager/Microsoft.Insights/preview/2017-05-01-preview/subscriptionDiagnosticsSettings_API.json#L323) has conflict with the [SubscriptionIdParameter](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/monitor/resource-manager/Microsoft.Insights/stable/2015-04-01/autoscale_API.json#L853) ， please unify its definition (use the second one) . Since the conflict already existed in the master branch ,  please ask @akning-ms to  force merge  after fixing the conflict
> 
> @jianyexi, @kairu-ms - the file you ask me to modify is owned by another resource provider (diagnosticSettings), it does not make any sense to force us to fix other RPs in order to be able to maintain our own. Can we skip this somehow ?

I am afraid can't, this will block the SDK generation process

**Comment by adklinge1** [comment]
@kairu-ms I fixed the linter conflict and run prettify on my files, can we merge the changes ?

**Review by JeffreyRichter — APPROVED** [review]

**Comment by kairu-ms** [comment]
> @kairu-ms I fixed the linter conflict and run prettify on my files, can we merge the changes ?

It's blocked by LintDiff issue.  Please ask @akning-ms to force merge.

**Comment by adklinge1** [comment]
> > @kairu-ms I fixed the linter conflict and run prettify on my files, can we merge the changes ?
> 
> It's blocked by LintDiff issue. Please ask @akning-ms to force merge.

 @kairu-ms , @jianyexi,  @akning-ms - it seems there is another linter problem blocking me now - 
 
image

Looking at the readme.ms, I can't tell what's wrong.

Can you please help clarifying what should be fixed now ?

**Comment by akning-ms** [comment]
for python SDK breaking change, @msyyc for confirm the impact. @ArcturusZhang to take a look GO SDK generation failure

**Comment by msyyc** [comment]
> for python SDK breaking change, @msyyc Yuchao Yan FTE for confirm the impact. @ArcturusZhang Dapeng Zhang FTE to take a look GO SDK generation failure

For python, the breaking change seems OK.

**Comment by ArcturusZhang** [comment]
> for python SDK breaking change, @msyyc for confirm the impact. @ArcturusZhang to take a look GO SDK generation failure

The pipeline looks like to be stucked, but I checked the log - the generation of go SDK succeeds and no breaking changes are introduced.

**Comment by adklinge1** [comment]
> > for python SDK breaking change, @msyyc for confirm the impact. @ArcturusZhang to take a look GO SDK generation failure
> 
> The pipeline looks like to be stucked, but I checked the log - the generation of go SDK succeeds and no breaking changes are introduced.

Thank youe @msyyc  + @ArcturusZhang. @kairu-ms  - do we have any other action items here or can we complete the PR ?

**Comment by kairu-ms** [comment]
@adklinge please resolve conflicts. Then I will help to merge it.

**Comment by adklinge1** [comment]
> @adklinge please resolve conflicts. Then I will help to merge it.

@kairu-ms  can you please help clarifying what are the conflicts?
I pulled from master and did not have any conflicts.

**Comment by kairu-ms** [comment]
@adklinge https://github.com/Azure/azure-rest-api-specs/pull/11898/conflicts

**Comment by adklinge** [comment]
> @adklinge https://github.com/Azure/azure-rest-api-specs/pull/11898/conflicts

Thanks! . Done

**Comment by kairu-ms** [comment]
Hi @jianyexi, this PR is blocked by ModelValidation errors. Can you help @adklinge solve them? Thanks.

**Comment by jianyexi** [comment]
> Hi @jianyexi Jianye Xi FTE, this PR is blocked by ModelValidation errors. Can you help @adklinge Adi Klinger FTE solve them? Thanks.

For the first model validation error, it's because there is an additional property: location in the example:  https://github.com/Azure/azure-rest-api-specs/blob/bb6a56f7283299dc3c6ed93c6b76cfdf990a33b4/specification/monitor/resource-manager/Microsoft.Insights/stable/2018-04-16/examples/createOrUpdateScheduledQueryRules.json#L8 , you should remove it

**Comment by adklinge** [comment]
> > Hi @jianyexi Jianye Xi FTE, this PR is blocked by ModelValidation errors. Can you help @adklinge Adi Klinger FTE solve them? Thanks.
> 
> For the first model validation error, it's because there is an additional property: location in the example:
> 
> https://github.com/Azure/azure-rest-api-specs/blob/bb6a56f7283299dc3c6ed93c6b76cfdf990a33b4/specification/monitor/resource-manager/Microsoft.Insights/stable/2018-04-16/examples/createOrUpdateScheduledQueryRules.json#L8
> 
> , you should remove it

Removed. There still are blocking validation tho

**Comment by adklinge** [comment]
> > Hi @jianyexi Jianye Xi FTE, this PR is blocked by ModelValidation errors. Can you help @adklinge Adi Klinger FTE solve them? Thanks.
> 
> For the first model validation error, it's because there is an additional property: location in the example:
> 
> https://github.com/Azure/azure-rest-api-specs/blob/bb6a56f7283299dc3c6ed93c6b76cfdf990a33b4/specification/monitor/resource-manager/Microsoft.Insights/stable/2018-04-16/examples/createOrUpdateScheduledQueryRules.json#L8
> 
> , you should remove it

@jianyexi  there are multiple validation errors claiming the "tags" and "location" properties are unexpected. However, both properties are defined in the model under "Resource", can you please help clarifying what's wrong ?

**Comment by raych1** [comment]
> > > Hi @jianyexi Jianye Xi FTE, this PR is blocked by ModelValidation errors. Can you help @adklinge Adi Klinger FTE solve them? Thanks.
> > 
> > 
> > For the first model validation error, it's because there is an additional property: location in the example:
> > https://github.com/Azure/azure-rest-api-specs/blob/bb6a56f7283299dc3c6ed93c6b76cfdf990a33b4/specification/monitor/resource-manager/Microsoft.Insights/stable/2018-04-16/examples/createOrUpdateScheduledQueryRules.json#L8
> > 
> > , you should remove it
> 
> @jianyexi there are multiple validation errors claiming the "tags" and "location" properties are unexpected. However, both properties are defined in the model under "Resource", can you please help clarifying what's wrong ?

@adklinge , the model validation failures are false alarm that are caused by duplicated model definition of `Resource` in both local swagger and referenced swagger from common-types folder. For short term solution, please suppress this error like [this](https://github.com/azure/azure-rest-api-specs/blob/master/specification/monitor/resource-manager/readme.md#suppression). Going forward in new api version, please remove local 'Resource' model and reference it from common-types folder.

**Pull request merged**
